### DISCUSSION TITLE
Flexget not seeing transmission-rpc up to date

### DISCUSSION THREAD

**Discussion by TiPi-Miou-Miou** [Flexget/Flexget#3800] [General]
Hi,

I've been using transmission/flexget on my rock64/Armbian for years (within a Nextcloudpi install).
To update my NexcloudPi I had to resinstall from scratch.
Then I had to re install Transmission & flexget.

Installation of Transmission and transmission-daemon & -rpc went well.
Installation of Flexget went well (somehow... some psutil issue (had to compile on the fly)))
I use my old Flexget config.yml file : 
```

templates:
  torrent:
    transmission:
      host: localhost
      port: 9091
      username: XXX
      password: XXX

tasks:
  tv_show:
    rss: http://some-rss-feed.com
    series:
      - Some Series Name
    template: torrent

schedules:
  - tasks: tv_show
    interval:
      hours: 1
```

My issue : Flexget is up to date. Transmission-rpc too (4.3). But when I try to test flexget config Flexget return an error regarding transmission, asking fo transmission-rpc 4.1.4<X<5.0.0.

my output messages : 
```
pi@nextcloudpi: pip install transmission-rpc -U
Defaulting to user installation because normal site-packages is not writeable
Requirement already satisfied: transmission-rpc in /usr/local/lib/python3.9/dist-packages (4.3.0)
Requirement already satisfied: requests<3.0.0,>=2.23.0 in /usr/lib/python3/dist-packages (from transmission-rpc) (2.25.1)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.9/dist-packages (from transmission-rpc) (4.7.0)

pi@nextcloudpi: flexget --test execute
2023-07-02 10:11:53 INFO     manager                       Test mode, creating a copy from database ...
2023-07-02 10:11:53 INFO     manager                       Test database created
2023-07-02 10:12:05 VERBOSE  task_queue                    There are 1 tasks to execute. Shutdown will commence when they have completed.
2023-07-02 10:12:06 CRITICAL transmission  tv_show         transmission-rpc module version >=4.1.4,<5.0.0 required.
2023-07-02 10:12:06 WARNING  task          tv_show         Aborting task (plugin: transmission)
2023-07-02 10:12:06 INFO     manager                       Removed test database

```
Where should I look ?
Do I have to change something in my config.yml file ?
Or in my Transmission.json file
Can I copy my flexget & transmission install from my old setup (still got a .img file of it) to my new (eventhough it's old version) ?

Thanks

**Comment by paranoidi** [comment]
You didn't mention how you have installed FlexGet. Is it on docker, git checkout, virtualenv or just as a root user. (Root user being least recommended). You need to have transmission-rpc installed in the same place where FlexGet is. Based on the log message "Defaulting to user installation because normal site-packages is not writeable" I think you have installed everything as root and then try to install transmission-rpc as a user and it falls back to installing it into pi user only. FlexGet will not find it from there.

I recommend installing FlexGet as a pi user into a virtualenv.

**Comment by TiPi-Miou-Miou** [comment]
Hi, Thanks for answering.

I followed the same [tutorial for installing](https://nothingbutsnark.svbtle.com/setting-up-a-raspberry-pi-to-torrent-a-tv-show-to-watch-through-xbmc) Transmission (Daemon, -rpc), Python, pip and flexget as before.
So as root.
But there was a problem installing flexget using this method (problem with psutil not beeing available)

So I followed the [official Flexget](https://flexget.com/InstallWizard/RPi4Lite) installation methode that uses pipx.
But still issue with psutil. So I installed gcc, psutil was compiled on the fly, flexget installed and there I am.
Now flexget is located here : /home/pi/.local/pipx/venvs/flexget/bin/flexget
But transmissionrpc installed in the python folder.

You say flexget and transmission-rpc should be installed in the same place ? Was not the case in my previous installation (I think)...
As flexget is installed in a venvs (I imagine the virtualenv you are talking about ?) how do I get transmission-rpc in the same place ? (I imagine simple "mv transmission-rpc" is not an option ? ^^')

Thanks

**Comment by paranoidi** [comment]
I see, the pipx installation method needs a bit different command to install dependencies. I added note to RPi4Lite page, but we should probably have it in every plugin page too...

You need to run: 

pipx inject flexget transmission-rpc

instead of the "pip install ..."

Sorry for the confusion

**Reply by TiPi-Miou-Miou** [reply]
Thanks a lot.
I check this out.
